src/dill/gauge-proxy-v2.sol

```diff
@@ -0,0 +1,2161 @@
+        address[] memory _tokenVote,
+        int256[] memory _weights,
+        uint256 _currentId
+    ) internal {
+        _reset(_owner, _currentId);
+        uint256 _tokenCnt = _tokenVote.length;
+        int256 _weight = int256(DILL.balanceOf(_owner));
+        int256 _totalVoteWeight = 0;
+        int256 _usedWeight = 0;
+
+        for (uint256 i = 0; i < _tokenCnt; i++) {
+            _totalVoteWeight += (_weights[i] > 0 ? _weights[i] : -_weights[i]);
+        }
+
+        for (uint256 i = 0; i < _tokenCnt; i++) {
+            address _token = _tokenVote[i];
+            address _gauge = gauges[_token];
+            int256 _tokenWeight = (_weights[i] * _weight) / _totalVoteWeight;
+            if (_gauge != address(0x0)) {
+                if (_currentId > tokenLastVotedPeriodId[_token]) {
+                    weights[_currentId][_token] = weights[
+                        tokenLastVotedPeriodId[_token]
+                    ][_token];
+
+                    tokenLastVotedPeriodId[_token] = _currentId;
+                }
+
+                weights[_currentId][_token] += _tokenWeight;
+                votes[_owner][_token] = _tokenWeight;
+                tokenVote[_owner].push(_token);
+
+                if (_tokenWeight < 0) _tokenWeight = -_tokenWeight;
+
+                _usedWeight += _tokenWeight;
+                totalWeight[_currentId] += _tokenWeight;
+            }
+        }
+        usedWeights[_owner] = _usedWeight;
+    }
+
+    // Vote with DILL on a gauge
+    function vote(address[] calldata _tokenVote, int256[] calldata _weights)
+        external
+    {
+        require(
+            _tokenVote.length == _weights.length,
+            "GaugeProxy: token votes count does not match weights count"
+        );
+        uint256 currentId = getCurrentPeriodId();
+        require(currentId > 0, "Voting not started yet");
+        _vote(msg.sender, _tokenVote, _weights, currentId);
+        delegations[msg.sender].blockDelegate[currentId] = true;
+    }
+
+    function setVotingDelegate(
+        address _delegateAddress,
+        uint256 _periodsCount,
+        bool _indefinite
+    ) external {
+        require(
+            _delegateAddress != address(0),
+            "GaugeProxyV2: cannot delegate zero address"
+        );
+        require(
+            _delegateAddress != msg.sender,
+            "GaugeProxyV2: delegate address cannot be delegating"
+        );
+
+        delegateData storage _delegate = delegations[msg.sender];
+
+        uint256 currentPeriodId = getCurrentPeriodId();
+
+        address currentDelegate = _delegate.delegate;
+        _delegate.delegate = _delegateAddress;
+        _delegate.prevDelegate = currentDelegate;
+        _delegate.updatePeriodId = currentPeriodId;
+
+        if (_indefinite == true) {
+            _delegate.indefinite = true;
+        } else if (_delegate.prevDelegate == address(0)) {
+            _delegate.endPeriod = currentPeriodId + _periodsCount - 1;
```

Comment:
Why is it that the `endPeriod` is subtracted by 1 week if there was no prior delegate?

Reply:
So if there is a previous delegate, the count for the end period of the next period starts from the next period as the prevDelegate will be active for the current period. If no previous delegate, just start the count from the current period.